### PR TITLE
Logs

### PR DESCRIPTION
**Pull request by noahingham**
Implements the following two delegate functions:

```go
SubscribeToLogs(chan do.Option)
UnsubscribeFromLogs(chan do.Option)
```

### CODE REVIEW

File: cmd/darknode/darknode.go [hunk 1]
```diff
@@ -38,6 +38,7 @@ func main() {
 	// Star the node.
 	time.Sleep(time.Second)
 	node.Start()
+	node.StartListening()
```
Comment (loongy) [inline cmd/darknode/darknode.go]: Remove this line, it's already in the cmd.

File: delegate.go [hunk 2]
```diff
@@ -78,6 +79,15 @@ func (node *DarkNode) OnBroadcastDeltaFragment(from identity.MultiAddress, delta
 		}
 		if delta.IsMatch(node.Configuration.Prime) {
 			log.Printf("%v compared (%v, %v) <- MATCH\n", node.Swarm.Address(), base58.Encode(deltaFragment.BuyOrderID), base58.Encode(deltaFragment.SellOrderID))
+			node.log(
+				"match",
+				fmt.Sprintf(
+					"{id: %s, buyID: %s, sellID: %s}",
```
Comment (loongy) [inline delegate.go]: This should be use the ` character so that you can format the string as JSON.

File: node.go [hunk 3]
```diff
@@ -27,6 +29,64 @@ var (
...
+	logQueue.channels = append(logQueue.channels, channel)
+}
+
+// Unsubscribe ...
+func (logQueue *LogQueue) Unsubscribe(channel chan do.Option) {
+	logQueue.Enter(nil)
+	defer logQueue.Exit()
+	length := len(logQueue.channels)
+	for i := 0; i < length; i++ {
+		if logQueue.channels[i] == channel {
```
Comment (loongy) [inline node.go]: Does channel equality testing work like this in go?

File: node.go [hunk 4]
```diff
@@ -42,6 +102,8 @@ type DarkNode struct {
 
 	quitServer chan struct{}
 	quitPacker chan struct{}
```
Comment (loongy) [inline node.go]: I don't think these channels are used anymore. Can we scrap them?

File: node.go (hunk 3 above)
Comment (noahingham) [reply node.go]: From the [Go spec](https://golang.org/ref/spec#Comparison_operators):

> Channel values are comparable. Two channel values are equal if they were created by the same call to make or if both have value nil.

File: node.go (hunk 4 above)
Comment (noahingham) [reply node.go]: Scrapped.

File: delegate.go (hunk 2 above)
Comment (noahingham) [reply delegate.go]: Fixed, thanks.

File: cmd/darknode/darknode.go (hunk 1 above)
Comment (noahingham) [reply cmd/darknode/darknode.go]: Removed.

File: node.go (hunk 3 above)
Comment (loongy) [reply node.go]: That's awesome. Sometime Go has nice surprises for us :)

**Pull request merged**